cmd/systray/systray.go

```diff
@@ -180,19 +181,36 @@ func (menu *Menu) eventLoop(ctx context.Context) {
 // watchIPNBus subscribes to the tailscale event bus and sends state updates to chState.
 // This method does not return.
 func watchIPNBus(ctx context.Context) {
+	for {
+		if err := watchIPNBusOnce(ctx); err != nil {
+			log.Println(err)
+			if errors.Is(err, context.Canceled) {
+				// If the context got canceled, we will never be able to
+				// reconnect to IPN bus, so exit the process.
+				log.Fatalf("watchIPNBus: %v", err)
+			}
+		}
+		// If our watch connection breaks, wait a bit before reconnecting. No
+		// reason to spam the logs if e.g. tailscaled is restarting or goes
+		// down.
+		time.Sleep(3 * time.Second)
+	}
+}
+
+func watchIPNBusOnce(ctx context.Context) error {
```

Comment:
nit: I'm not sure of a better name, but this makes it sounds like a `sync.Once` that is **really** only ever executed once.  It's unexported, so not really THAT big of a deal, but just a tad confusing.

Reply:
that's fair, renamed to `watchIPNBusInner` :shrug: